modules/images/webp-uploads/image-edit.php

```diff
@@ -19,18 +19,41 @@
  */
 function webp_uploads_update_sources( $metadata, $valid_mime_transforms, $main_images, $subsized_images ) {
 	foreach ( $valid_mime_transforms as $targeted_mime ) {
-		// Make sure the path and file exists as those values are being accessed.
-		if ( ! isset( $main_images[ $targeted_mime ]['path'], $main_images[ $targeted_mime ]['file'] ) || ! file_exists( $main_images[ $targeted_mime ]['path'] ) ) {
-			continue;
+		// Make sure the path and file exists as those values are required.
+		$image_directory = null;
+		if ( isset( $main_images[ $targeted_mime ]['path'], $main_images[ $targeted_mime ]['file'] ) && file_exists( $main_images[ $targeted_mime ]['path'] ) ) {
+			// Add sources to original image metadata.
+			$metadata['sources'][ $targeted_mime ] = array(
+				'file'     => $main_images[ $targeted_mime ]['file'],
+				'filesize' => filesize( $main_images[ $targeted_mime ]['path'] ),
+			);
+			$image_directory                       = pathinfo( $main_images[ $targeted_mime ]['path'], PATHINFO_DIRNAME );
 		}
 
-		$image_directory = pathinfo( $main_images[ $targeted_mime ]['path'], PATHINFO_DIRNAME );
+		/**
+		 * If no original image was provided the image_directory can't be determined, in that scenario try to
+		 * find it from the `file` property.
+		 *
+		 * @see get_attached_file()
+		 */
+		if (
+			null === $image_directory
+			&& isset( $metadata['file'] )
+			&& 0 !== strpos( $metadata['file'], '/' )
+			&& ! preg_match( '|^.:\\\|', $metadata['file'] )
```

Comment:
Does it check whether the file path is Windows-based? Like `C:\...`?  I think the regular expression is incorrect because it has three back slashes instead of two.
```suggestion
			&& ! preg_match( '|^.:\\|', $metadata['file'] )
```

Or even better to just check the substring of the file path:
```suggestion
			&& ':\' !== substr( $metadata['file'], 1, 2 )
```

Reply:
> Does it check whether the file path is Windows-based? Like C:\...? I think the regular expression is incorrect because it has three back slashes instead of two.

Looks like that's the case, however this might be an existing bug in WP core as well as this code comes from core `get_attached_file` will replace with the `substr` instead. 

https://github.com/WordPress/wordpress-develop/blob/5.9/src/wp-includes/post.php#L725

Reply:
Actually is because the `\` is escaped that's why it needs 3 if you need an expression with 2 back slashs.

Reply:
> Actually is because the `\` is escaped that's why it needs 3 if you need an expression with

Ah... okay, thanks for clarifying... makes sense. 👍